pkg/ddl/ddl_worker.go

```diff
@@ -1357,11 +1372,22 @@ func (w *worker) runDDLJob(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64,
 		err = dbterror.ErrInvalidDDLJob.GenWithStack("invalid ddl job type: %v", job.Type)
 	}
 
+	// there are too many job types, instead let every job type output its own
+	// updateRawArgs, we try to use these rules as a generalization:
+	//
+	// if job has no error, some arguments may be changed, there's no harm to update
+	// it.
+	updateRawArgs = err == nil
+	// if job changed from running to rolling back, arguments may be changed
+	if prevState == model.JobStateRunning && job.IsRollingback() {
+		updateRawArgs = true
```

Comment:
we should move runtime args change in something like job ctx, we actively mark it if those args changed and fill back depends on the mark

now we are checking whether args are changed passively.

Reply:
Yes I started in the active way yesterday, and found there are too many cases and many function signatures need to be changed 😂